src/main/java/backend/resource/MultiModel.java

```diff
@@ -49,6 +49,12 @@ private synchronized MultiModel add(Model model) {
         return this;
     }
 
+    public synchronized MultiModel removeRepoModelById(String RepoId) {
+        Optional<Model> repoModelToBeDeleted = getModelById(RepoId);
+        this.models.remove(repoModelToBeDeleted.get().getRepoId());
```

Comment:
You need to check if the value is present here, otherwise this can fail at runtime (like dereferencing a null pointer).

I got an email with a comment where you said this wasn't working. Is that resolved?